### PR TITLE
Add env var overrides

### LINKED ISSUE #38: Log bucket storing data indefinitely
I'm not very convinced that we need our deployment test results for very long. The log bucket seems to be storing data indefinitely.

https://github.com/byu-oit/terraform-aws-postman-test-lambda/blob/fd8af3d6e7e58be253419a93b2cc71a68832f755/main.tf#L55-L58

Here are some screenshots from today over in an S3 bucket used by hw-fargate-api:
to_delete
deleted

### PR DESCRIPTION
**Pull request by blakemorgan**
This still needs to be tested, but would add a way to override env vars. Making PR for a POC. Also fixes #38 and merges the other open PRs.

- Update dependabot
- Update deps
- Add test env var override variable
- Add env vars function
- Bump version and package
- Delete data after 4 months

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -24,9 +24,10 @@ locals {
   ])
   using_local_files = length(local.local_collections) + length(local.local_environments) > 0
   lambda_env_variables = {
-    S3_BUCKET           = local.using_local_files ? aws_s3_bucket.postman_bucket[0].bucket : null
-    POSTMAN_COLLECTIONS = jsonencode(var.postman_collections)
-    POSTMAN_API_KEY     = var.postman_api_key
+    S3_BUCKET              = local.using_local_files ? aws_s3_bucket.postman_bucket[0].bucket : null
+    POSTMAN_COLLECTIONS    = jsonencode(var.postman_collections)
+    POSTMAN_API_KEY        = var.postman_api_key
+    TEST_ENV_VAR_OVERRIDES = base64encode(jsonencode(var.test_env_var_overrides))
```
Comment (yoshutch) [inline main.tf]: (nit) curious why base64 encoding the override variables?

File: lambda/src/index.js [hunk 2]
```diff
@@ -170,6 +171,16 @@ async function updateRunner (deploymentId, combinedRunner, event, error) {
   }
 }
 
+function generateEnvVars() {
+  const envVarsArray = []
+  const parsedVars = JSON.parse(Buffer.from(process.env.TEST_ENV_VAR_OVERRIDES, 'base64'))
+  if (Object.keys(parsedVars).length === 0) return envVarsArray
```
Comment (mzroth) [inline lambda/src/index.js:177]: I take it by this part, if nothing is passed in as env vars, it won't override anything (ie it won't be a breaking change). Is that right?

Comment (blakemorgan) [reply lambda/src/index.js:177]: Yes

File: main.tf (hunk 1 above)
Comment (blakemorgan) [reply main.tf]: I was over thinking it.

File: lambda/src/index.js [hunk 3]
```diff
@@ -171,9 +171,9 @@ async function updateRunner (deploymentId, combinedRunner, event, error) {
   }
 }
 
-function generateEnvVars() {
+function generateEnvVars () {
   const envVarsArray = []
-  const parsedVars = JSON.parse(Buffer.from(process.env.TEST_ENV_VAR_OVERRIDES, 'base64'))
+  const parsedVars = JSON.parse(process.env.TEST_ENV_VAR_OVERRIDES, 'base64')
```
Comment (yoshutch) [inline lambda/src/index.js:176]: I'm guessing you don't need to parse base64 now, right?

**Comment by mzroth** [comment]
LGTM when it has been tested.

**Review by mzroth — DISMISSED** [review]

File: lambda/src/index.js (hunk 3 above)
Comment (blakemorgan) [reply lambda/src/index.js:176]: yes

**Comment by blakemorgan** [comment]
This has been tested and works as expected.

**Review by yoshutch — APPROVED** [review]

**Pull request merged**